quiz/models.py

```diff
@@ -0,0 +1,75 @@
+from django.db import models
+from common.models import Class, Subject, User, Semester
+
+
+# Model that represents a quiz
+class Quiz(models.Model):
+    name = models.CharField(max_length=100)
+    # root directory of stored quiz
+    root = models.CharField(max_length=255, default="quizzes")
+    # relative quiz path of quiz root directory
+    src = models.CharField(max_length=255, verbose_name="Directory", unique=True)
+    subject = models.ForeignKey(Subject, on_delete=models.CASCADE)
+    semester = models.ForeignKey(Semester, on_delete=models.CASCADE)
+    created_at = models.DateTimeField(auto_now_add=True)
+
+    def __str__(self):
+        return self.name
+
+    class Meta:
+        verbose_name_plural = "Quizzes"
+
+
+# Model that represents an assigned quiz
+class AssignedQuiz(models.Model):
+    quiz = models.ForeignKey(Quiz, on_delete=models.CASCADE)
+    clazz = models.ForeignKey(Class, on_delete=models.CASCADE)
+    assigned = models.DateTimeField()
+    # duration of quiz in minutes
+    duration = models.IntegerField()
```

Comment:
I can imagine long running quizzes that don't have a duration, but we can always make it nullable in the future, so let's keep it required for now.